src/Streamly/Internal/Data/SVar/Worker.hs

```diff
@@ -0,0 +1,544 @@
+-- |
+-- Module      : Streamly.Internal.Data.SVar.Worker
+-- Copyright   : (c) 2017 Composewell Technologies
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+--
+module Streamly.Internal.Data.SVar.Worker
+    (
+    -- * Adjusting Limits
+      decrementYieldLimit
+    , incrementYieldLimit
+    , decrementBufferLimit
+    , incrementBufferLimit
+    , resetBufferLimit
+
+    -- * Rate Control
+    , Work (..)
```

Comment:
Should `WorkInfo` also be in this module?

Reply:
No it is used in the SVar type definition so has to be in SVar.Type.